src/placement/jump_map.c

```diff
@@ -409,8 +415,39 @@ get_target(struct pool_domain *curr_dom, struct pool_target **target,
 			end_dom = start_dom + (num_doms - 1);
 
 			range_set = isset_range(dom_used, start_dom, end_dom);
-			if (range_set)
-				clrbit_range(dom_used, start_dom, end_dom);
+			if (range_set) {
+				int idx;
+
+				/* Skip the domain whose targets are used up */
```

Comment:
So we'll never support colocated shard anymore? No matter if they are in different RDG? For instance, a pool map with 4 domains and 1 target each domain,  for a RP_2G2, should we allow colocated shard when selecting spare target?

Reply:
you mean we can not do colocated within same vos target? yes, we can not do it with current implementation anyway (there is a lot needs to be fixed). If we decide to do, we can fix it in later patch.

Reply:
What are the existing problems of colocating two shards (in different RDGs) on same VOS target?

Reply:
I think we should add this in a separate patch that supports this directly. The current implementation doesn't do anything to correctly place different redundancy groups once all targets have been filled - it just clears all the maps and hopes for the best. This might be at the worst time, placing replica 1 on the last target, then placing replica 2 on the same target. A proper implementation of that would clear the tgts_used array pre-emptively when finishing the previous group when there isn't enough space left for the next redundancy group to ensure that redundancy guarantees are still being met.